packages/mobile/src/exchange/ExchangeReview.tsx

```diff
@@ -66,13 +70,32 @@ export class ExchangeReview extends React.Component<Props, State> {
   }
 
   onPressConfirm = () => {
-    const makerToken = this.state.makerToken
+    const { makerToken, inputToken, inputAmount } = this.state
     const makerAmount = this.getMakerAmount()
+    // BEGIN: Analytics
+    const isDollarToGold = inputToken === CURRENCY_ENUM.DOLLAR
```

Comment:
This seems like a lot of data wrangling just for analytics. Does the saga do any of this already? If so, how about we trigger analytics from there instead?

Reply:
Well, it does not calculate quite, so I've decided to have it there. Essentially it does a couple `get` from state and a couple of arithmetic calculations, so should not be too heavy.